### PR TITLE
feat: upgrade to react 18 - fix build #8

### PR DESCRIPTION
**Pull request by neatbyte-vnobis**
## Changes
Upgraded version of react-dnd.

## How Has This Been Tested?
Manually.

## Documentation
None.

### CODE REVIEW

**Comment by neatbyte-ivelychko** [comment]
@brunozoric this pr is ready for review
I just need to fix some `any` types.

File: packages/app-form-builder/src/admin/views/Editor.tsx [hunk 1]
```diff
@@ -9,7 +9,6 @@ const EditorView = () => {
          * react-dnd users old version of React which has children defined by default.
          * This will break build when react-dnd is upgraded, so then we can remove ts-expect-error
```
Comment (brunozoric) [inline packages/app-form-builder/src/admin/views/Editor.tsx]: Feel free to remove the comment :)

File: packages/app-headless-cms/src/admin/components/Droppable.tsx [hunk 2]
```diff
@@ -33,10 +33,15 @@ export interface DroppableProps {
     onDrop?: OnDropCallable;
 }
 
+type DragObject = {
+    item: any;
```
Comment (brunozoric) [inline packages/app-headless-cms/src/admin/components/Droppable.tsx]: Really no type for item that can be used?

File: packages/app-headless-cms/src/admin/components/Droppable.tsx [hunk 3]
```diff
@@ -33,10 +33,15 @@ export interface DroppableProps {
...
+    isOver: boolean;
+};
+
 const DroppableComponent = (props: DroppableProps) => {
     const { children, onDrop, isVisible = () => true } = props;
 
-    const [{ item, isOver }, drop] = useDrop({
+    const [{ item, isOver }, drop] = useDrop<unknown, unknown, DragObject>({
```
Comment (brunozoric) [inline packages/app-headless-cms/src/admin/components/Droppable.tsx]: What are those `unknown`s defining when you send them to the `useDrop`?

File: packages/app-headless-cms/src/admin/components/FieldEditor/FieldEditorContext.tsx [hunk 4]
```diff
@@ -231,15 +235,18 @@ export const FieldEditorProvider = ({
         return null;
     }, []);
 
-    const onEndDrag: OnEndDragCallable = ({ type, field, fields }, monitor) => {
+    const onEndDrag: OnEndDragCallable<unknown, DropResult> = (
```
Comment (brunozoric) [inline packages/app-headless-cms/src/admin/components/FieldEditor/FieldEditorContext.tsx]: What does the `unknown` define?

File: packages/app-page-builder/src/editor/plugins/toolbar/navigator/navigatorHooks.ts [hunk 5]
```diff
@@ -58,23 +58,16 @@ interface UseSortableListArgs {
...
-    beginDrag,
-    endDrag
-}: UseSortableListArgs) => {
+export const useSortableList = ({ index, move, type, beginDrag, endDrag }: UseSortableListArgs) => {
     const ref = useRef<HTMLDivElement>(null);
     const [dropItemAbove, setDropItemAbove] = useState(false);
     const isDraggingDownwardsRef = useRef<boolean>(false);
 
-    const [dropData, drop] = useDrop({
+    const [dropData, drop] = useDrop<DragItem, unknown, CollectedProps>({
```
Comment (brunozoric) [inline packages/app-page-builder/src/editor/plugins/toolbar/navigator/navigatorHooks.ts]: What does the `unknown` define?

File: packages/app-page-builder/src/types.ts [hunk 6]
```diff
@@ -214,6 +219,11 @@ export type PbElementDataType = {
     [key: string]: any;
 };
 
+export interface CollectedProps {
+    handlerId: string | symbol | null;
+    isOver: any;
```
Comment (brunozoric) [inline packages/app-page-builder/src/types.ts]: No better type for `isOver`?

**Review by brunozoric — CHANGES_REQUESTED** [review]

File: packages/app-headless-cms/src/admin/components/Droppable.tsx (hunk 2 above)
Comment (neatbyte-ivelychko) [reply packages/app-headless-cms/src/admin/components/Droppable.tsx]: @brunozoric `item` described as `any` in all interfaces inside of Droppable.tsx

I will try to find a better type.
Maybe it should be described as `DragSource`.

P.S. Fixed type

File: packages/app-page-builder/src/editor/plugins/toolbar/navigator/navigatorHooks.ts (hunk 5 above)
Comment (neatbyte-ivelychko) [reply packages/app-page-builder/src/editor/plugins/toolbar/navigator/navigatorHooks.ts]: First two Generic arguments of `useDrop<DragObject, DropResult>` and `useDrag<DragObject, DropResult>` are being used to defined types for the methods of `useDrop` and `useDrag`. 
`DropResult` defines what should methods like `hover` for `useDrop` or `end` for `useDrag` return.
In most scenarios we are calling custom functions inside of `end` or `hover` and those custom functions retun `void`.
`DragObject` defines what should mehods of `monitor` return.

File: packages/app-page-builder/src/types.ts (hunk 6 above)
Comment (neatbyte-ivelychko) [reply packages/app-page-builder/src/types.ts]: Should be boolean

**Review by brunozoric — APPROVED** [review]

**Pull request merged**